### PR TITLE
chore: Move limits to *uint64 and add optional markers

### PR DESCRIPTION
**Pull request by xiu**
## Description

References:
* https://github.com/prometheus-operator/prometheus-operator/pull/5658#discussion_r1227145581
* https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

n/a

### CODE REVIEW

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -1346,14 +1346,14 @@ func generateRunningFilter() yaml.MapSlice {
...
 	if enforced != nil {
-		if user < *enforced && user != 0 || *enforced == 0 {
-			return user
+		if user != nil && *user < *enforced && *user != 0 || *enforced == 0 {
+			return *user
 		}
 		return *enforced
 	}
-	return user
+	return 0
```
Comment (simonpasquier) [inline pkg/prometheus/promcfg.go:1356]: wouldn't it be better to return nil and the caller would set the field only if it's not nil?

Comment (xiu) [reply pkg/prometheus/promcfg.go]: @ArthurSens Good point, if we now return a *uint64, we can do that.

Comment (xiu) [reply pkg/prometheus/promcfg.go]: Updated in 17c9d21d9218a6f48a3c5918f735f5ea916a4adc

File: pkg/prometheus/promcfg.go [hunk 2]
```diff
@@ -205,16 +205,21 @@ var (
 
 // AddLimitsToYAML appends the given limit key to the configuration if
 // supported by the Prometheus version.
-func (cg *ConfigGenerator) AddLimitsToYAML(cfg yaml.MapSlice, k limitKey, limit uint64, enforcedLimit *uint64) yaml.MapSlice {
-	if limit == 0 && enforcedLimit == nil {
+func (cg *ConfigGenerator) AddLimitsToYAML(cfg yaml.MapSlice, k limitKey, limit *uint64, enforcedLimit *uint64) yaml.MapSlice {
+	if limit == nil && enforcedLimit == nil {
+		return cfg
+	}
```
Comment (simonpasquier) [inline pkg/prometheus/promcfg.go:211]: (nit) `getLimit()` already covers this case?

```suggestion
```

Comment (xiu) [reply pkg/prometheus/promcfg.go]: true! updated :)

**Review by simonpasquier — APPROVED** [review]
lgtm! I'll defer to @ArthurSens who's taking care of the release if this should merge or wait :)

**Review by JoaoBraveCoding — APPROVED** [review]
lgtm

**Comment by ArthurSens** [comment]
> lgtm! I'll defer to @ArthurSens who's taking care of the release if this should merge or wait :)

Somehow I missed this message 😅 

Thanks @xiu for tackling this one, it makes life easier for the new contributor :)

**Pull request merged**